packages/frontend-v2/components/AddCourseModal/AddCourseModal.tsx

```diff
@@ -74,8 +74,15 @@ export const AddCourseModal: React.FC<AddCourseModalProps> = ({
 
     const updatedSelectedCourses = [...selectedCourses];
 
-    // push course + any coreqs of the course
-    const coreqs = await getRequiredCourseCoreqs(course);
+    // grab any coreqs of the course that haven't already been selected/added to the term
+    const coreqs = (await getRequiredCourseCoreqs(course)).filter((coreq) => {
+      const isAlreadySelected = selectedCourses.find((selectedCourse) =>
+        isEqualCourses(selectedCourse, coreq)
+      );
+      const isAlreadyAdded = isCourseInCurrTerm(coreq);
+      return isAlreadyAdded || isAlreadySelected;
```

Comment:
I think this condition is reversed at the moment (only keeps courses that are already added or already selected). I tested it with the suggested change and it works as intended.

```suggestion
      return !(isAlreadyAdded || isAlreadySelected);
```

Reply:
Thank you so so much, I seriously need to stop doing graduate dev at work, my brain is making some very dumb mistakes